packages/ckeditor5-dev-tests/bin/check-dependencies.js

```diff
@@ -13,134 +13,164 @@ const glob = require( 'glob' );
-			missingPackages.devDependencies
-				.map( entry => '- ' + entry )
-				.join( '\n' ),
-
-			// Unused dependencies.
-			// We need to remove packages which are already defined as `missing`.
-			// `unused.dependencies` lists packages used by JS files. It does not touch CSS files where
-			// a package can be used.
-			unused.dependencies
-				.filter( entry => missingPackages.dependencies.includes( entry ) )
-				.map( entry => '- ' + entry )
-				.join( '\n' ),
-
-			// Unused devDependencies.
-			// We need to remove packages listed as `unused.dependencies` in order to avoid duplicating them.
-			// Also, we need to filter `unused.devDependencies`. See `unused.dependencies` in order to know why.
-			unused.devDependencies
-				.filter( entry => missingPackages.dependencies.includes( entry ) )
-				.filter( packageName => !unused.dependencies.includes( packageName ) )
-				.map( entry => '- ' + entry )
-				.join( '\n' ),
-
-			// Relative CSS imports (files do not exist).
-			nonExistingCSSFiles
-				.map( entry => {
-					return `- "${ chalk.italic( entry.file ) }" imports "${ chalk.italic( entry.import ) }"`;
-				} )
-				.join( '\n' ),
-
-			// Duplicated `dependencies` and `devDependencies`.
-			findDuplicatedDependencies( packageJson.dependencies, packageJson.devDependencies )
-				.map( entry => '- ' + entry )
-				.join( '\n' )
-		];
-
-		const hasErrors = data.some( entry => !!entry );
-
-		if ( !hasErrors ) {
-			console.log( chalk.green( 'All dependencies are defined correctly.' ) );
-
-			return;
-		}
+checkDependencies( packagePaths );
 
-		console.log( chalk.red( 'Found some issue with dependencies.\n' ) );
+/**
+ * Checks dependencies sequentially in all provided packages.
+ *
+ * @param {Set.<String>} packagePaths Relative paths to packages.
+ */
+async function checkDependencies( packagePaths ) {
+	for ( const packagePath of packagePaths ) {
+		const isSuccess = await checkDependenciesInPackage( packagePath );
 
-		if ( data[ 0 ] ) {
-			console.log( chalk.yellow( 'Invalid itself imports found in:' ) );
-			console.log( data[ 0 ] + '\n' );
-			console.log( '🧾 Imports from local package must always use relative path.\n' );
+		// Mark result of this script execution as invalid.
+		if ( !isSuccess ) {
+			process.exitCode = 1;
 		}
+	}
+}
 
-		if ( data[ 1 ] ) {
-			console.log( chalk.red( 'Missing dependencies:' ) );
-			console.log( data[ 1 ] + '\n' );
-		}
+/**
+ * Checks dependencies in provided package. If the folder does not contain a package.json file the function quits with success.
+ *
+ * @param {String} packagePath Relative path to package.
+ */
+async function checkDependenciesInPackage( packagePath ) {
+	const packageAbsolutePath = path.resolve( packagePath );
+	const packageJsonPath = path.join( packageAbsolutePath, 'package.json' );
 
-		if ( data[ 2 ] ) {
-			console.log( chalk.red( 'Missing devDependencies:' ) );
-			console.log( data[ 2 ] + '\n' );
-		}
+	if ( !fs.existsSync( packageJsonPath ) ) {
```

Comment:
I'd display a warning message here. There must be something wrong with the current setup if the `package.json` file is missing.

Reply:
Warning has been added.

The CI calls `ls packages -1` to list all packages. The reason why I added this condition here is that after building the DLL, in the `packages/` path 4 new directories are created:

- `packages/ckeditor5-dll-balloon`
- `packages/ckeditor5-dll-classic`
- `packages/ckeditor5-dll-decoupled-document`
- `packages/ckeditor5-dll-inline`

which do not contain `package.json` files (they are just the output from the DLL build procedure and they contain only the built editor).

Reply:
I guess we removed those packages a few days ago. Check whether you can delete those from your env.
Anyway, the warning will help with finding invalid packages so it's good to have it.